### PR TITLE
ntp-388: Cookies page created and google analytics enabled

### PR DESCRIPTION
**Pull request by stephenjmcneill**
## Context

Cookies page created and google analytics enabled

## Changes proposed in this pull request

Cookie page created
End to End testing added
Google analytics enabled and disabled

Cookies page

## Guidance to review

Check google banner disappears when no cookie page. 
That enabling /disabling of google analytics can be done from cooker banner and cookie page.
The user can return previous page before accessing cookie statement.

## Link to Jira ticket

https://dfedigital.atlassian.net/browse/NTP-388

## Things to check

- [x] Title is in the format "NTP-XXX: Short description" where NTP-XXX is the Jira ticket reference
- [x] Code and tests follow the [coding standards](/docs/coding-standards.md)
- [ ] `dotnet format` has been run in the repository root
- [ ] Test coverage of new code is at least 80%
- [x] All UI related acceptance criteria specified in the ticket have been added to the relevant [cypress test feature file(s)](/UI/cypress/e2e/)
- [ ] Unless explicitly mentioned in the ticket, all UI work should have been tested as working on desktop and mobile resolutions with JavaScript on and off
- [ ] Database migrations can be applied to the current codebase in main without causing exceptions
- [ ] Debug logging has been added after all logic decision points
- [ ] All [automated testing](/README.md#testing) including accessibility and security has been run against your code

### CODE REVIEW

File: UI/Pages/Cookies.cshtml [hunk 1]
```diff
@@ -1,10 +1,134 @@
...
+                        <div class="govuk-radios__item">
+                            <input class="govuk-radios__input" data-testid="cookie-consent-deny" id="cookie-consent-deny" name="consent" type="radio" value="false" checked="@(Model.Consent == false)">
+                            <label class="govuk-label govuk-radios__label" for="cookie-consent-deny">
+                                No, do not use cookies to track my website usage
+                            </label>
+                        </div>
+                    </div>
+                </fieldset>
+            </div>
+            <input type="submit" data-testid="save-changes" value="Save changes" class="govuk-button" data-module="govuk-button">
```
Comment (benagain) [inline UI/Pages/Cookies.cshtml:84]: The [content doc](https://docs.google.com/document/d/1UUkpfkyhyg12aQDF17--H0eMRYB3IlMmmh7hT3QA2SI/edit#heading=h.ns60qtm90eqj) appears to want the button to be labelled "Save my choice"

image

File: UI/Pages/Cookies.cshtml [hunk 2]
```diff
@@ -1,10 +1,134 @@
...
+                <tr>
+                    <td class="govuk-table__cell">_gid</td>
+                    <td class="govuk-table__cell">Checks if you’ve signed in to / accessed the service before. This helps us count how many people use it.</td>
+                    <td class="govuk-table__cell">24 hours</td>
+                </tr>
+            </tbody>
+        </table>
+        <h2 class="govuk-heading">Essential cookies</h2>
+        <p class="govuk-body">
+            These cookies need to be on to allow you to use this service. They don’t store your personal data. For more information read our <a data-testid="privacy-page" asp-page="./Privacy" class="govuk-link" data-testid="privact-policy-link">privacy policy</a><br />.
```
Comment (benagain) [inline UI/Pages/Cookies.cshtml:110]: Is the `<br/>` at the end of line necessary?

File: UI/Pages/Cookies.cshtml.cs [hunk 3]
```diff
@@ -7,15 +7,77 @@ namespace UI.Pages
...
+        public string? returnPath { get; set; }
+
+        public IActionResult OnGet(bool? consent, string returnUrl, bool? viewCookieStatement)
         {
+            returnPath = returnUrl;
+
+            if (Request.Cookies.ContainsKey(ConsentCookieName))
+            {
+                Consent = bool.Parse(Request.Cookies[ConsentCookieName]);
+            }
```
Comment (benagain) [inline UI/Pages/Cookies.cshtml.cs:21]: There are a couple of null reference warnings that I think would be good to resolve

File: UI/Pages/Cookies.cshtml.cs [hunk 4]
```diff
@@ -7,15 +7,77 @@ namespace UI.Pages
     public class Cookies : PageModel
     {
         private const string ConsentCookieName = ".FindATuitionPartner.Consent";
-        public IActionResult OnGet(bool? consent, string returnUrl)
+        public bool? Consent { get; set; }
+        public bool PreferencesSet { get; set; } = false;
+        public string? returnPath { get; set; }
+
+        public IActionResult OnGet(bool? consent, string returnUrl, bool? viewCookieStatement)
```
Comment (benagain) [inline UI/Pages/Cookies.cshtml.cs:14]: Is `viewCookieStatement` used anywhere?

File: UI/Pages/Cookies.cshtml.cs [hunk 5]
```diff
@@ -7,15 +7,77 @@ namespace UI.Pages
...
+
+        private void ApplyCookieConsent(bool? consent)
+        {
+            if (consent.HasValue)
+            {
+                var cookieOptions = new CookieOptions { Expires = DateTime.Today.AddMonths(12), Secure = true };
+                Response.Cookies.Append(ConsentCookieName, consent.Value.ToString(), cookieOptions);
+            }
+
+            if (!consent.Value)
```
Comment (benagain) [inline UI/Pages/Cookies.cshtml.cs:71]: Will this throw an exception if `consent` is null?

File: UI/Pages/Cookies.cshtml (hunk 1 above)
Comment (stephenjmcneill) [reply UI/Pages/Cookies.cshtml:84]: Changed

File: UI/Pages/Cookies.cshtml.cs (hunk 4 above)
Comment (stephenjmcneill) [reply UI/Pages/Cookies.cshtml.cs:14]: Removed

File: UI/Pages/Cookies.cshtml.cs (hunk 5 above)
Comment (stephenjmcneill) [reply UI/Pages/Cookies.cshtml.cs:71]: Null now checked

**Review by benagain — APPROVED** [review]

**Pull request merged**